### ISSUE TITLE
Clock widget - date format

### ISSUE THREAD

**Issue by mad-ady** [home-assistant/appdaemon#187]
I'm looking to change the format of the date shown in the clock widget. I've found the code responsible for it and it shows:
```
self.set_field(self, "date", today.toLocaleDateString());
```
The javascript function toLocaleDateString() should do it, but for me (Chrome under Linux), it's not doing anything (still displays the US format).

I've tried the following (the ro_RO.UTF-8 locale is available on my system):
* start chrome with ```LANG=ro_RO.UTF-8 chromium-browser```
* start chrome with ```LC_TIME=ro_RO.UTF-8 chromium-browser```
* apply this in Chrome settings (set Romanian language as preferred): https://stackoverflow.com/questions/37221494/how-to-change-the-locale-in-chrome-browser

In all tests the following javascript code returns the same (US) date format:
```
today = new Date();
Wed Dec 27 2017 17:33:47 GMT+0200 (EET)
today.toLocaleDateString();
"12/27/2017"
```
I'd like to know if there is something that I can do to influence the correct locale in my browser, or if it would be better/acceptable to change the clock code and allow a user-configurable date format?

Thanks

**Comment by acockburn** [comment]
It seems like a problem in Chrome - the best workaround it seems would be to add formatting to the widget. I can add that to my list or I'd be happy to accept a Pull Request.

**Comment by mad-ady** [comment]
Thank you for the confirmation. I'll try to work something out and send you
a PR.

**Comment by mad-ady** [comment]
I've played with it a bit, and it seems that toLocaleDateString() is good enough for most needs, especially since javascript doesn't have a native implementation of strftime(). So, if you pass a country code to toLocaleDateString - e.g. date.toLocateDateString("ro"), it will output the correct format.

I've added this code change (I can't seem to create a different pull request just for this, so here is the commit on my fork): https://github.com/mad-ady/appdaemon/commit/83d5543cd68afc4ca74604175f7029876857f44f

It needs a documentation change as well, adding an optional parameter to the clock widget where you can add a country code:

```
clock:
    widget_type: clock
    time_format: 24hr
    date_format_country: "ro"
    show_seconds: 0
    time_style: "color: yellow; font-size: 40pt; font-weight: bold;"
    date_style: "font-size: 16pt; font-weight: bold;"
```

There are further things which can be added. toLocaleDateString also can take an attributes object where you can customize how each item is displayed. Like weekdays shown as names, etc. Here's an example (not implemented because I was lazy):

```
d = new Date();
Date 2018-01-07T12:41:33.795Z
d.toLocaleDateString();
"1/7/2018"
d.toLocaleDateString("ro");
"07.01.2018"
d.toLocaleDateString("ro", {"weekday": "long", "day": "numeric", "year": "numeric" });
"2018 duminică 7"
d.toLocaleDateString("de", {"weekday": "long", "day": "numeric", "month": "long"});
"Sonntag, 7. Januar"
```
I've also sent this as a PR, but it seems bundled with some previous commits. Not sure if you can cherrypick.

**Comment by mad-ady** [comment]
Also added this commit: https://github.com/home-assistant/appdaemon/commit/6de5ea04a4450b9a31809fc8f5cf5eda65511e03 to the baseclock so that you can customize the way the date looks as described in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString. Its bundled with the previous PR - apparently github does it automatically.

Regarding configuration, here is an example:
```
clock:
    widget_type: clock
    time_format: 24hr
    date_format_country: "ro"
    date_format_options:
      weekday: "short"
      day: "numeric"
      month: "numeric"
    show_seconds: 0
    time_style: "color: yellow; font-size: 40pt; font-weight: bold;"
    date_style: "font-size: 20pt; font-weight: bold;"
```
The date_format_options depends on date_format_country being set (will not apply without it).
Here's a rendered example:
https://imgur.com/uUt3KwR.png

**Comment by acockburn** [comment]
OK, merged and documented - will appear on 3.0b3.
